pages/api/models/component.ts

```diff
@@ -0,0 +1,32 @@
+import { model, Schema, Types } from "mongoose";
+
+enum ComponentType {
+    TEXT = "text",
+    VIDEO = "video",
+    AUDIO = "audio",
+}
+
+interface ComponentInterface {
+    _id: Types.ObjectId;
+    type: ComponentType;
+    properties: any;
+}
+
+const ComponentSchema = new Schema<ComponentInterface>(
```

Comment:
we decided not to make the properties a required field?

Reply:
it might make more sense to make properties a required field since a component will be somewhere on the screen, meaning it has at least position properties